lib/array-slice.js

```diff
@@ -65,13 +65,25 @@ var ArraySlice = createClass({
    * @param callback - Function to execute for each element.
    * @param thisArg - Value to use as this (i.e the reference Object) when executing callback
    * @returns {Element}
+   * @deprecated
    * @memberof ArraySlice.prototype
    */
   flatMap: function (callback, thisArg) {
+    return this.compactMap(callback, thisArg);
```

Comment:
We should remove this completely, since we are still pre 1.0, we can afford to break it and need no deprecation periods.

Reply:
I want to remove it in the next minor release, but given how hard it is to make a minor release (we have to release entire minim and fury toolchain for breaking changes) then I would put this in next patch release of Minim and remove the deprecated method in the next minor release (when that happens).

I'd wait with minor release until after the differentiator is ran incase we have further fixes to make to the JSON 0.6 logic.

Reply:
Given we have https://github.com/refractproject/minim/pull/175 open, I'd suggest we merge this PR as-is and then I'll rebase #175 and make those tests pass. So `flatMap` behaves like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap. Since current behaviour is something else, modelled from `flatMap` from other languages.